pageserver/src/repository.rs

```diff
@@ -119,24 +123,30 @@ pub trait Timeline: Send + Sync {
     fn gc_iteration(&self, horizon: u64) -> Result<GcResult>;
 }
 
-pub trait History: Iterator<Item = Result<RelationUpdate>> {
+pub trait History: Iterator<Item = Result<Modification>> {
     /// The last_valid_lsn at the time of history() call.
     fn lsn(&self) -> Lsn;
 }
 
+//
+// Structure representing any update operation of object storage.
+// It is used to copy object storage content in PUSH method.
+//
 #[derive(Debug, PartialEq, Eq, Serialize, Deserialize)]
-pub struct RelationUpdate {
-    pub rel: RelTag,
+pub struct Modification {
```

Comment:
It seems like this has become the tuple (key excluding timeline & lsn, lsn, value)

We had moved away from sending raw key values over the network, and instead opted for an abstract format that could be used across different repository types.

Reply:
Sorry, I do not completely understand you. So we replace RelTag with ObjectTag. What is principally changed?
What do you mean by "raw key values"and "abstract format"? Key is serialized using some serializer (explicitely written or generates, as in our case). At low level we always sending over network raw sequence of bytes. But at higher level we know how to interpret them. I.e. it is our protocol. What do you suggest to use here?